pkg/image/api/types.go

```diff
@@ -60,6 +84,69 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureState holds a result of signature's server-side check.
+type SignatureState string
+
+// SignatureForImage describes relation of the signature to the image containing it.
+type SignatureForImage string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side signature check. IOW a time of the last update of SignatureState.
+	Checked *unversioned.Time
+	// Says whether the signature matches image object containing it.
+	ForImage SignatureForImage
+	// Result of the last server-side signature check performed at Checked. Filled by server.
+	State SignatureState
+
+	// Following metadata fields will be set by server if the signature content is successfully parsed and
+	// the information available.
+
+	// Digest of manifest of the signed image (e.g. "sha256:xxxxx...").
+	ImageHash string
+	// A human readable string representing image's identity. It could contain company's name, a product name
+	// and a version. Eligible value is image's pull specification
+	// (e.g. "registry.access.redhat.com/rhel7/rhel:7.2").
+	ImageIdentity string
+	// Optional. Contains interesting signed claims not important enough to become primary fields.
+	SignedClaims map[string]string
+	// If specified, it is the time of signature's creation.
+	Created *unversioned.Time
+	// If specified, it holds information about an issuer of signing certificate or key (a person or entity
+	// who signed the signing certificate or key).
+	IssuedBy SignatureIssuer
+	// If specified, it holds information about a subject of signing certificate or key (a person or entity
+	// who signed the image).
+	IssuedTo SignatureSubject
+}
+
+// SignatureGenericEntity holds a generic information about a person or entity who is an issuer or a subject
+// of signing certificate or key.
+type SignatureGenericEntity struct {
+	// Organization name.
+	Organization string
+	// Common name (e.g. openshift-signing-service).
+	CommonName string
+}
+
+// SignatureIssuer holds information about an issuer of signing certifikate or key.
+type SignatureIssuer struct {
+	SignatureGenericEntity
+}
+
+// SignatureSubject holds information about a person or entity who created the signature.
+type SignatureSubject struct {
+	SignatureGenericEntity
+	// If present, it is a fingerprint of public key belonging to the subject used to verify image signature.
+	PublicKeyID string
+	// If present, it is an algorithm used to hash the public key to create its ID.
+	PublicKeyIDAlgorithm string
```

Comment:
I'm not sure we need this since a client that is smart enough to double check it, it's smart enough to parse the signature right?

`PublicKeyID` makes sense I think.